chained-batch.js

```diff
@@ -1,25 +1,32 @@
 const util = require('util')
 const AbstractChainedBatch = require('abstract-leveldown').AbstractChainedBatch
+const binding = require('./binding')
 
 function ChainedBatch (db) {
   AbstractChainedBatch.call(this, db)
-  this.binding = db.binding.batch()
+  this.context = binding.batch_init(db.context)
 }
 
 ChainedBatch.prototype._put = function (key, value) {
-  this.binding.put(key, value)
+  binding.batch_put(this.context, key, value)
 }
 
 ChainedBatch.prototype._del = function (key) {
-  this.binding.del(key)
+  binding.batch_del(this.context, key)
 }
 
 ChainedBatch.prototype._clear = function () {
-  this.binding.clear()
+  binding.batch_clear(this.context)
 }
 
 ChainedBatch.prototype._write = function (options, callback) {
-  this.binding.write(callback)
+  // TODO (ensure docs covers the following)
+  // Note that we're passing in options here, which we didn't do before. We
+  // must do this so we can use the `sync` property, which we didn't handle before
+  // since we never passed in an object at time of creation (bug) (the previous c++
+  // used to assume we did this from the js side from the ChainedBatch()
+  // constructor).
+  binding.batch_write(this.context, options, callback)
```

Comment:
This needs documentation. @vweevers Maybe write an issue for this before this is released?

Reply:
#541